go/vt/vttablet/tabletmanager/rpc_vreplication.go

```diff
@@ -44,3 +49,73 @@ func (tm *TabletManager) VReplicationExec(ctx context.Context, query string) (*q
 func (tm *TabletManager) VReplicationWaitForPos(ctx context.Context, id int32, pos string) error {
 	return tm.VREngine.WaitForPos(ctx, id, pos)
 }
+
+// UpdateVRWorkflow updates the sidecar databases's vreplication
+// record for this tablet's vreplication workflow stream.
+// Note: the VReplication engine creates a new controller for the
+// workflow when the record is updated, so we also in effect restart
+// the workflow via the update.
+func (tm *TabletManager) UpdateVRWorkflow(ctx context.Context, req *tabletmanagerdatapb.UpdateVRWorkflowRequest) (*tabletmanagerdatapb.UpdateVRWorkflowResponse, error) {
+	query := "select id, source, cell, tablet_types from %s.vreplication where workflow = %a"
+	bindVars := map[string]*querypb.BindVariable{
+		"wf": sqltypes.StringBindVariable(req.Workflow),
+	}
+	parsed := sqlparser.BuildParsedQuery(query, sidecardb.GetIdentifier(), ":wf")
+	stmt, err := parsed.GenerateQuery(bindVars, nil)
+	if err != nil {
+		return nil, err
+	}
+	res, err := tm.VREngine.Exec(stmt)
+	if err != nil {
+		return nil, err
+	}
+	if res == nil || len(res.Rows) == 0 {
+		return nil, fmt.Errorf("no stream found for workflow %s", req.Workflow)
+	}
+
+	row := res.Named().Row()
+	id := row.AsInt64("id", 0)
+	cells := row.AsString("cell", "")
+	tabletTypes := row.AsString("tablet_types", "")
+	bls := &binlogdatapb.BinlogSource{}
+	source := row.AsBytes("source", []byte{})
+	// For the string values, we use NULL to differentiate from
+	// an empty string. The NULL value indicates that we should
+	// keep the existing value.
+	if req.Cells != sqltypes.NULL.String() {
+		cells = req.Cells
+	}
+	if req.TabletTypes != sqltypes.NULL.String() {
+		tabletTypes = req.TabletTypes
+	}
+	if err = prototext.Unmarshal(source, bls); err != nil {
+		return nil, err
+	}
+	// If we don't want to update the existing value pass -1 or any
+	// other invalid value.
+	if _, ok := binlogdatapb.OnDDLAction_name[int32(req.OnDdl)]; ok {
+		bls.OnDdl = req.OnDdl
+	}
+	source, err = prototext.Marshal(bls)
+	if err != nil {
+		return nil, err
+	}
+	query = "update %s.vreplication set source = %a, cell = %a, tablet_types = %a where id = %a"
+	bindVars = map[string]*querypb.BindVariable{
+		"sc": sqltypes.StringBindVariable(string(source)),
+		"cl": sqltypes.StringBindVariable(cells),
+		"tt": sqltypes.StringBindVariable(tabletTypes),
+		"id": sqltypes.Int64BindVariable(id),
+	}
+	parsed = sqlparser.BuildParsedQuery(query, sidecardb.GetIdentifier(), ":sc", ":cl", ":tt", ":id")
```

Comment:
`ParseAndBind()` will make your life happier.

Reply:
I use it when I can.